packages/js/data/src/crud/types.ts

```diff
@@ -13,3 +22,79 @@ export type Item = {
 export type ItemQuery = BaseQueryParams & {
 	[ key: string ]: unknown;
 };
+
+export type CrudActions< ResourceName, ItemConfig, MutableProperties > =
+	MapActions<
+		{
+			create: ( query: ItemQuery ) => Item;
+			update: ( query: ItemQuery ) => Item;
+		},
+		ResourceName,
+		MutableProperties,
+		ItemConfig
```

Comment:
I think this might have to be `Generator< unknown, ItemConfig >` and same for line 42 below for the `delete` action.
Without this I would get `void` as the return type for `actions.deleteProductAttribute( 2 );`, but with this I get `Promise<ProductAttribute>` as the return type, which would be the correct one.
I believe the `DispatchFromMap` the `CrudActions` get piped into (in the product-attributes store) looks if the return type is a generator and if so, it will return a Promise, otherwise `void` as default.

Reply:
Nice!  Updated in https://github.com/woocommerce/woocommerce/pull/33680/commits/89e430c07ade8d8af7b0ffb777c804bb477eaca7.